assets/js/base/context/cart-checkout/shipping/index.js

```diff
@@ -147,14 +165,27 @@ export const ShippingDataProvider = ( { children } ) => {
 				EMIT_TYPES.SHIPPING_RATE_SELECT_FAIL,
 				currentErrorStatus
 			);
-		} else if ( ! isSelectingRate && selectedRates ) {
+		}
+	}, [
+		selectedRates,
+		isSelectingRate,
+		currentErrorStatus,
+		shippingErrorStatus,
+	] );
+
+	useEffect( () => {
+		if (
+			! isSelectingRate &&
+			selectedRates &&
+			! currentErrorStatus.hasError
+		) {
```

Comment:
Similar to elsewhere, I found splitting out the emitters into their own effects helps reduce the amount of times the emitters (and effects) fire.